### PR TITLE
Miscellaneous cli-related parity fixes

### PR DESCRIPTION
**Pull request by coolreader18**
Part of this is taken from #5414.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -183,33 +172,46 @@ fn run_rustpython(vm: &VirtualMachine, run_mode: RunMode, quiet: bool) -> PyResu
...
+        );
+
+        eprintln!("Type \"help\", \"copyright\", \"credits\" or \"license\" for more information.");
+    }
+    let res = match run_mode {
         RunMode::Command(command) => {
             debug!("Running command {}", command);
-            vm.run_code_string(scope, &command, "<stdin>".to_owned())?;
+            vm.run_code_string(scope.clone(), &command, "<stdin>".to_owned())
+                .map(drop)
```
Comment (youknowone) [inline src/lib.rs:196]: Is this a common convention to drop value from Result? It looks like `drop` is important logic here to me. If not, a weak suggestion.

```suggestion
                .map(|_| ())
```

File: src/settings.rs [hunk 2]
```diff
@@ -358,8 +345,13 @@ fn settings_from(matches: &ArgMatches) -> (Settings, RunMode) {
...
 /// Get environment variable and turn it into integer.
-fn get_env_var_value(name: &str) -> Result<u8, std::env::VarError> {
-    env::var(name).map(|value| u8::from_str(&value).unwrap_or(1))
+fn get_env_var_value(name: &str) -> Option<u8> {
+    env::var_os(name).filter(|v| !v.is_empty()).map(|value| {
+        value
+            .to_str()
+            .and_then(|v| v.parse::<u8>().ok())
+            .unwrap_or(1)
+    })
```
Comment (youknowone) [inline src/settings.rs:354]:
```suggestion
    let value_str = env::var_os(name).filter(|v| !v.is_empty())?;
    let value = value_str
        .to_str()
        .and_then(|v| v.parse::<u8>().ok())
        .unwrap_or(1);
    Some(value)
```

**Review by youknowone — APPROVED** [review]

File: src/settings.rs (hunk 2 above)
Comment (coolreader18) [reply src/settings.rs:354]: Given there's 2 layers of Option here, I think I'd like to keep the logic as it is - looking at the suggested version my instinct was to change it to `.to_str()?.parse().unwrap_or(1)`, but that's different. I think the explicit `map()` makes more clear the control flow and conditions.

File: src/lib.rs (hunk 1 above)
Comment (coolreader18) [reply src/lib.rs:196]: I've seen `.map(drop)` a decent amount, yeah.

File: src/settings.rs (hunk 2 above)
Comment (youknowone) [reply src/settings.rs:354]: line 349 and the following lines are unrelated logics. early return reveals line 350 is independent to 349 and code readers can forget about 349 when going to 350.
Using `map` make readers essentially follow the full logic without benefit.

**Pull request merged**